usaspending_api/etl/management/commands/elasticsearch_indexer.py

```diff
@@ -0,0 +1,234 @@
+        #   And keep it timezone-award for S3
+        config["starting_date"] = get_last_load_date(f"es_{options['load_type']}", default=default_datetime)
+
+    config["max_query_size"] = settings.ES_TRANSACTIONS_MAX_RESULT_WINDOW
+    if options["load_type"] == "awards":
+        config["max_query_size"] = settings.ES_AWARDS_MAX_RESULT_WINDOW
+
+    config["is_incremental_load"] = not bool(config["create_new_index"]) and (
+        config["starting_date"] != default_datetime
+    )
+
+    if config["is_incremental_load"]:
+        write_alias = settings.ES_TRANSACTIONS_WRITE_ALIAS
+        if config["load_type"] == "awards":
+            write_alias = settings.ES_AWARDS_WRITE_ALIAS
+        if config["index_name"]:
+            logger.info(format_log(f"Ignoring provided index name, using alias '{write_alias}' for incremental load"))
+        config["index_name"] = write_alias
+        if not es_client.cat.aliases(name=write_alias):
+            logger.error(format_log(f"Write alias '{write_alias}' is missing"))
+            raise SystemExit(1)
+    else:
+        if es_client.indices.exists(config["index_name"]):
+            logger.error(format_log(f"Data load into existing index. Change index name or run an incremental load"))
+            raise SystemExit(1)
+
+    if not config["directory"].is_dir():
+        logger.error(format_log(f"Provided directory does not exist"))
+        raise SystemExit(1)
+    elif config["starting_date"] < default_datetime:
+        logger.error(format_log(f"--start-datetime is too early. Set no earlier than {default_datetime}"))
+        raise SystemExit(1)
+    elif not config["is_incremental_load"] and config["process_deletes"]:
+        logger.error(format_log("Skipping deletions for ths load, --deleted overwritten to False"))
+        config["process_deletes"] = False
+
+    config["ingest_wait"] = options["idle_wait_time"]
+
+    return config
+
+
+def set_config(copy_args: list, arg_parse_options: dict) -> dict:
+    """Set values based on env vars and when the script started"""
+    root_index = settings.ES_TRANSACTIONS_QUERY_ALIAS_PREFIX
+    if arg_parse_options["load_type"] == "awards":
+        root_index = settings.ES_AWARDS_QUERY_ALIAS_PREFIX
+    config = {
+        "aws_region": settings.USASPENDING_AWS_REGION,
+        "s3_bucket": settings.DELETED_TRANSACTION_JOURNAL_FILES,
+        "root_index": root_index,
+        "processing_start_datetime": datetime.now(timezone.utc),
+        "verbose": arg_parse_options["verbosity"] > 1,  # convert the management command's levels of verbosity to a bool
+    }
+
+    config.update({k: v for k, v in arg_parse_options.items() if k in copy_args})
+    return config
+
+
+def fiscal_years_for_processing(options: list) -> list:
+    if "all" in options["fiscal_years"]:
+        return create_fiscal_year_list(start_year=parse_fiscal_year(settings.API_SEARCH_MIN_DATE))
+    return [int(x) for x in options["fiscal_years"]]
```

Comment:
Something I noticed when you don't do the `all` FY option, is it (nicely) processes fiscal years in reverse. Typically the more relevant/hot data that people are looking for are going to be from the latest FY. Is this the place that would order the FYs to process, and if so can it be made to start with the latest FY and work backward?

If a failure happens late in the process, it increases the odds that the more-important data loaded before the failure.

Reply:
Really? I thought the fiscal-years arg was required. Do you mean providing a list of FYs in descending order processes the ETL in the same order?

I expect future changes to DEV-5856 will significantly reduce the significance of fiscal years order in the processing.